### PR TITLE
Implement COMMENT ON COLUMN in MySQL

### PR DESCRIPTION
**Pull request by agrawalreetika**
Implement COMMENT ON COLUMN in MySQL #5333

### CODE REVIEW

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java [hunk 1]
```diff
@@ -365,6 +365,19 @@ public boolean isLimitGuaranteed(ConnectorSession session)
...
+    @Override
+    public void setColumnComment(JdbcIdentity identity, JdbcTableHandle handle, JdbcColumnHandle column, Optional<String> comment)
+    {
+        String sql = format(
+                "ALTER TABLE %s MODIFY %s %s(%s) COMMENT %s",
+                quoted(handle.getRemoteTableName()),
+                column.getColumnName(),
+                column.getJdbcTypeHandle().getJdbcTypeName().get(),
+                column.getJdbcTypeHandle().getColumnSize(),
+                format("'%s'", comment.get()));
```
Comment (ebyhr) [inline presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java:377]: This logic will remove some existing column information (e.g. `DEFAULT`). We should keep such information.

File: presto-mysql/src/test/java/io/prestosql/plugin/mysql/BaseMySqlIntegrationSmokeTest.java [hunk 2]
```diff
@@ -263,6 +263,9 @@ public void testColumnComment()
                 "SELECT column_name, comment FROM information_schema.columns WHERE table_schema = 'tpch' AND table_name = 'test_column_comment'",
                 "VALUES ('col1', 'test comment'), ('col2', null), ('col3', null)");
 
+        assertUpdate("COMMENT ON COLUMN tpch.test_column_comment.col1 IS 'test comment in col1'");
+        assertUpdate("COMMENT ON COLUMN tpch.test_column_comment.col1 IS ''");
```
Comment (ebyhr) [inline presto-mysql/src/test/java/io/prestosql/plugin/mysql/BaseMySqlIntegrationSmokeTest.java:267]: Please add test case for `IS NULL` and assertion to make sure exactly the column comment was set.

**Review by ebyhr — CHANGES_REQUESTED** [review]

**Comment by ebyhr** [comment]
Please update `TestMySqlDistributedQueries` that has `supportsCommentOnColumn`.

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java (hunk 1 above)
Comment (agrawalreetika) [reply presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java:377]: @ebyhr Yes, I missed on this. Currently with this command, it's overriding existing values. The same command with original column information should help in retaining existing column information. Though I could find only a nullable check from existing objects but not any other column info, could you please help me with your input, what logic can I use here to get the column original definition?

Comment (ebyhr) [reply presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java:377]: You will be able to get the column definition from JDBC `DatabaseMetaData.getColumns`.

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java [hunk 3]
```diff
@@ -365,6 +365,39 @@ public boolean isLimitGuaranteed(ConnectorSession session)
         return true;
     }
 
+    @Override
+    public void setColumnComment(JdbcIdentity identity, JdbcTableHandle handle, JdbcColumnHandle column, Optional<String> comment)
+    {
+        try (Connection connection = connectionFactory.openConnection(identity)) {
+            ResultSet resultSet = getColumns(handle, connection.getMetaData());
+
+            String colDefintion = "";
```
Comment (ebyhr) [inline presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java:374]: Please avoid abbreviation. 
```suggestion
            String columnDefinition = "";
```

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java [hunk 4]
```diff
@@ -365,6 +365,39 @@ public boolean isLimitGuaranteed(ConnectorSession session)
...
+                    break;
+                }
+            }
+            resultSet.close();
+
+            String sql = format(
+                    "ALTER TABLE %s MODIFY %s COMMENT %s",
+                    quoted(handle.getRemoteTableName()),
+                    colDefintion,
+                    comment.isPresent() ? format("'%s'", comment.get()) : "''");
```
Comment (ebyhr) [inline presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java:393]: We can simplify as `"ALTER TABLE %s MODIFY %s COMMENT '%s'"` and `comment.orElse("")`

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java [hunk 5]
```diff
@@ -365,6 +365,39 @@ public boolean isLimitGuaranteed(ConnectorSession session)
...
+    public void setColumnComment(JdbcIdentity identity, JdbcTableHandle handle, JdbcColumnHandle column, Optional<String> comment)
+    {
+        try (Connection connection = connectionFactory.openConnection(identity)) {
+            ResultSet resultSet = getColumns(handle, connection.getMetaData());
+
+            String colDefintion = "";
+            while (resultSet.next()) {
+                if (resultSet.getString("COLUMN_NAME").equalsIgnoreCase(column.getColumnName())) {
+                    colDefintion = format("%s %s(%s)%s%s%s",
+                            column.getColumnName(),
```
Comment (ebyhr) [inline presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java:378]: The generated query will fail if the column is reserved word.

**Review by ebyhr — COMMENTED** [review]
Could you please add more test cases to make sure the detailed column definition?

File: presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java (hunk 5 above)
Comment (agrawalreetika) [reply presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java:378]: @ebyhr Could you please provide an example on this?

File: presto-mysql/src/test/java/io/prestosql/plugin/mysql/BaseMySqlIntegrationSmokeTest.java (hunk 2 above)
Comment (agrawalreetika) [reply presto-mysql/src/test/java/io/prestosql/plugin/mysql/BaseMySqlIntegrationSmokeTest.java:278]: @ebyhr  test case is added for IS NULL. Could you please provide me an example for the second part?

Comment (ebyhr) [reply presto-mysql/src/test/java/io/prestosql/plugin/mysql/BaseMySqlIntegrationSmokeTest.java:278]: We can add utility method for directly getting `ResultSet` from MySQL and then assert fields in `information_schema.columns`. 


```java
    private ResultSet executeQuery(String sql)
            throws SQLException
    {
        try (Connection connection = DriverManager.getConnection(mysqlServer.getJdbcUrl(), mysqlServer.getUsername(), mysqlServer.getPassword());
                Statement statement = connection.createStatement()) {
            return statement.executeQuery(sql);
        }
    }
```

**Comment by colebow** [comment]
:wave: @agrawalreetika - this PR is inactive and doesn't seem to be under development, and it might already be implemented. If you'd like to continue work on this at any point in the future, feel free to re-open.

**Pull request closed without merging**